core-android/src/main/java/com/uber/sdk/android/core/utils/AppProtocol.java

```diff
@@ -10,52 +9,108 @@
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.util.Base64;
+import android.util.Pair;
+import com.uber.sdk.android.core.SupportedAppType;
+
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
-import java.util.HashSet;
+import java.util.*;
+
+import static com.uber.sdk.android.core.SupportedAppType.UBER;
+import static com.uber.sdk.android.core.SupportedAppType.UBER_EATS;
 
 public class AppProtocol {
+    /**
+     * @deprecated Use {@link #getSupportedPackageNames()}.
+     */
+    @Deprecated
     public static final String[] UBER_PACKAGE_NAMES =
-            {"com.ubercab", "com.ubercab.presidio.app", "com.ubercab.presidio.exo",
-                    "com.ubercab.presidio.development"};
+            {"com.ubercab", "com.ubercab.presidio.app", "com.ubercab.presidio.exo", "com.ubercab.presidio.development"};
+
+    private static final String[] RIDER_PACKAGE_NAMES =
+            {"com.ubercab.presidio.development", "com.ubercab.presidio.exo", "com.ubercab.presidio.app", "com.ubercab"};
+    private static final String[] EATS_PACKAGE_NAMES =
+            {"com.ubercab.eats.debug", "com.ubercab.eats.exo", "com.ubercab.eats"};
```

Comment:
These constant arrays were created in reverse of the original public one since the method using it required a change in the implementation.  This maintains the behavior where debug/exo builds are preferred over the production build.